tests/lib/filesystem.py

```diff
@@ -0,0 +1,47 @@
+"""Helpers for filesystem-dependent tests.
+"""
+import os
+import socket
+import subprocess
+import sys
+from functools import partial
+from itertools import chain
+
+from .path import Path
+
+
+def make_socket_file(path):
+    # Socket paths are limited to 108 characters (sometimes less) so we
+    # chdir before creating it and use a relative path name.
+    cwd = os.getcwd()
+    os.chdir(os.path.dirname(path))
+    try:
+        sock = socket.socket(socket.AF_UNIX)
+        sock.bind(os.path.basename(path))
+    finally:
+        os.chdir(cwd)
+
+
+def make_unreadable_file(path):
+    Path(path).touch()
+    os.chmod(path, 0o000)
+    if sys.platform == "win32":
+        username = os.environ["USERNAME"]
```

Comment:
Hmm... IIUC, `os.getlogin()` can get the username without needing this environment variable on Windows. Let's use that here and avoid exposing the `USERNAME` to the tests.

https://github.com/python/cpython/blob/c4cacc8c5eab50db8da3140353596f38a01115ca/Modules/posixmodule.c#L7112-L7120

Reply:
`os.getlogin` is only available on Unix platforms in Python 2. While the general change being made is Python 3-only, I think this utility function should be version-agnostic. I can add a comment to that effect here and in the `tox.ini` (including string PY2) so we know to come back to it and switch after dropping support.

Reply:
Updated [here](https://github.com/pypa/pip/pull/6844/files#diff-d26b0fd18cee791716156c2b48f4213aR29) and [here](https://github.com/pypa/pip/pull/6844/files#diff-b91f3d5bd63fcd17221b267e851608e8R13).